scripts/vsphere/attach_vm

```diff
@@ -160,12 +162,95 @@ write_config_context_vm() {
 
   write_config_context
 
-
   echo "export VM_APP=${VM_APP}" >>${CONTEXT_FILE}
   echo "export VM_IP=${VM_IP}" >>${CONTEXT_FILE}
 
   rm ".attachConfig"
   cat ${OUTPUT_DIR}/vm-${VM_APP}/".context"
 }
 
+# multi nic support
+multinicdetection() {
+  # cache all VM ip table info for later use
+  IP_INFO=$(ssh -i "${KEY_FILE}" -o StrictHostKeyChecking=no root@${VM_IP} ip -j a )
+
+  NICCOUNTSTRING=$(echo "${IP_INFO}" | jq -jr '.[].ifname| type')
+  NICSTRING=$(echo "${IP_INFO}" | jq -jr '.[]|" ",.ifname')
+
+  # Case 1: VM has only one nic. fill in MANAGED_ADDR and MANAGED_NIC if unspecified.
+  if [[ ${#NICCOUNTSTRING} == 12 ]]; then
```

Comment:
can you explain what is the meaning of the magic number 12?

Reply:
NICCOUNTSTRING is the count for how many nic on VM with concatenating "string" which is of length6. 
 if a VM has more than 2 nic like {lo ens192 ens224}, NICCOUNTSTRING="stringstringstring". User need to specify at least a --managed-nic or --managed-address.